### PR TITLE
Move GetFedoraToolsGuestAgentBlacklistUserData

### PR DESCRIPTION
**Pull request by sradco**
Moved GetFedoraToolsGuestAgentBlacklistUserData out of /test/utils.go to /tests/libnet/cloudinit/cloudinit.go

### What this PR does
Before this PR:

After this PR:

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by sradco** [comment]
@iholder101 is there an issue with the tests? I think the 2 tests that fail are not related to the PR, but they keep failing..

**Comment by oshoval** [comment]
> ✋🧢
> 
> /hold
> 
> Dear @sradco
> 
> ⚠️ this pull request exceeds the number of retests that are allowed per individual commit.
> 
> 🔎 Please check that the changes you committed are fine and that there are no infrastructure issues present!
> 
> 👌 After all issues have been resolved, you can remove the hold on this pull request by commenting `/unhold` on it.
> 
> 🙇 Thank you, your friendly referee automation, on behalf of the @sig-buildsystem and the KubeVirt community!

It might be a bug that the hold bot calculate manual retests ? 
@dhiller can you please suggest ?

**Comment by dhiller** [comment]
> > ✋🧢
> > /hold
> > Dear @sradco
> > ⚠️ this pull request exceeds the number of retests that are allowed per individual commit.
> > 🔎 Please check that the changes you committed are fine and that there are no infrastructure issues present!
> > 👌 After all issues have been resolved, you can remove the hold on this pull request by commenting `/unhold` on it.
> > 🙇 Thank you, your friendly referee automation, on behalf of the @sig-buildsystem and the KubeVirt community!
> 
> It might be a bug that the hold bot calculate manual retests ? @dhiller can you please suggest ?

No, that's by intent.

**Comment by dhiller** [comment]
@iholder101 @xpivarc do you have time to look into the failures and get to the ground why it is failing?

**Comment by sradco** [comment]
@dhiller How can I proceed with this PR at this point?

**Review by codingben — APPROVED** [review]
I reviewed your code changes and don't see something wrong (including commit message). Hopefully we can get CI working again.

/lgtm

**Comment by iholder101** [comment]
> @oshoval it might make sense to distinguish between manual retests and automatic ones, and to generate a more reasonable comment in each case. However, the comment about that retesting without any changes still makes sense.

@dhiller If someone does `/test <specific-test>` I guess it's not being counted, right? If so, I'd argue that the current behavior is actually desired.

Generally speaking, there is no reason to manually `/retest-required` or `/retest`. If tests fail it's reasonable that the CI is under pressure, and it won't help if every PR owner would re-trigger the tests manually again and again, but rather wait for the bot to trigger them. Using `/test <specific>` is different, since the PR owner sometimes have a need to examine that something specific works without triggering the whole suit.

**Comment by iholder101** [comment]
> /retest-required

@sradco please avoid manually re-trigerring the tests in the future, but instead wait for the bot to re-trigger them for you. This causes CI pressure and cancels currently running jobs.

Saying that, this PR brings a pure refactoring, so I don't see a reason any behavior would change.
/unhold

File: tests/libnet/cloudinit/cloudinit.go [hunk 1]
```diff
@@ -199,3 +199,9 @@ func CreateDefaultCloudInitNetworkData() string {
 	}
 	return data
 }
+
+func GetFedoraToolsGuestAgentBlacklistUserData(commands string) string {
+	return fmt.Sprintf(`#!/bin/bash
+            echo -e "\n\nBLACKLIST_RPC=%s" | sudo tee -a /etc/sysconfig/qemu-ga
+`, commands)
+}
```
Comment (iholder101) [inline tests/libnet/cloudinit/cloudinit.go:207]: Can we use the builder pattern defined in this file to achieve the same?
Should we convert it to `WithSomething()`?

Comment (oshoval) [reply tests/libnet/cloudinit/cloudinit.go:207]: no need imo, the usage is very specific and not common

Comment (iholder101) [reply tests/libnet/cloudinit/cloudinit.go:207]: Cool, thank you for the feedback.
Otherwise everything looks good to me :+1: 

/approve

**Comment by dhiller** [comment]
> > @oshoval it might make sense to distinguish between manual retests and automatic ones, and to generate a more reasonable comment in each case. However, the comment about that retesting without any changes still makes sense.
> 
> @dhiller If someone does `/test <specific-test>` I guess it's not being counted, right? If so, I'd argue that the current behavior is actually desired.
> 
> Generally speaking, there is no reason to manually `/retest-required` or `/retest`. If tests fail it's reasonable that the CI is under pressure, and it won't help if every PR owner would re-trigger the tests manually again and again, but rather wait for the bot to trigger them. Using `/test <specific>` is different, since the PR owner sometimes have a need to examine that something specific works without triggering the whole suit.

@iholder101 IIUC you are suggesting to not count `test` commands in general towards the retest count? Since technically there is no difference between a `test all` and a `retest` we had decided not to distinguish between them when counting retests. My comment above was just suggesting that the comment should be different to the user issuing the command, if the user is not a bot user.

**Comment by iholder101** [comment]
> > > @oshoval it might make sense to distinguish between manual retests and automatic ones, and to generate a more reasonable comment in each case. However, the comment about that retesting without any changes still makes sense.
> > 
> > 
> > @dhiller If someone does `/test <specific-test>` I guess it's not being counted, right? If so, I'd argue that the current behavior is actually desired.
> > Generally speaking, there is no reason to manually `/retest-required` or `/retest`. If tests fail it's reasonable that the CI is under pressure, and it won't help if every PR owner would re-trigger the tests manually again and again, but rather wait for the bot to trigger them. Using `/test <specific>` is different, since the PR owner sometimes have a need to examine that something specific works without triggering the whole suit.
> 
> @iholder101 IIUC you are suggesting to not count `test` commands in general towards the retest count? Since technically there is no difference between a `test all` and a `retest` we had decided not to distinguish between them when counting retests.

Right, `test all` is a special case, and I agree it should be counted.
But what about `/test <specific>`? is it being counted as well? I think this might be an exception that's not being counted. I think on draft PRs that test a specific lane before being ready for review, for example.

> My comment above was just suggesting that the comment should be different to the user issuing the command, if the user is not a bot user.

I can agree with that. I think that the message should encourage people to avoid manual testing. This generally a bad idea.

**Comment by dhiller** [comment]
> > > > @oshoval it might make sense to distinguish between manual retests and automatic ones, and to generate a more reasonable comment in each case. However, the comment about that retesting without any changes still makes sense.
> > > 
> > > 
> > > @dhiller If someone does `/test <specific-test>` I guess it's not being counted, right? If so, I'd argue that the current behavior is actually desired.
> > > Generally speaking, there is no reason to manually `/retest-required` or `/retest`. If tests fail it's reasonable that the CI is under pressure, and it won't help if every PR owner would re-trigger the tests manually again and again, but rather wait for the bot to trigger them. Using `/test <specific>` is different, since the PR owner sometimes have a need to examine that something specific works without triggering the whole suit.
> > 
> > 
> > @iholder101 IIUC you are suggesting to not count `test` commands in general towards the retest count? Since technically there is no difference between a `test all` and a `retest` we had decided not to distinguish between them when counting retests.
> 
> Right, `test all` is a special case, and I agree it should be counted. But what about `/test <specific>`? is it being counted as well? I think this might be an exception that's not being counted. I think on draft PRs that test a specific lane before being ready for review, for example.

Makes sense to not count specific `/test <whatever>` against the retest count. TBH I didn't look to closely at the implementation, I'll look at it soon.

> 
> > My comment above was just suggesting that the comment should be different to the user issuing the command, if the user is not a bot user.
> 
> I can agree with that. I think that the message should encourage people to avoid manual testing. This generally a bad idea.

**Comment by brianmcarey** [comment]
/override pull-kubevirt-e2e-k8s-1.29-sig-compute-migrations

Lane upgraded to pull-kubevirt-e2e-k8s-1.30-sig-compute-migrations which passed

**Pull request merged**